include/vikunja/mem/iterator/ConstantIterator.hpp

```diff
@@ -0,0 +1,267 @@
+ * This file is part of vikunja.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/alpaka.hpp>
+
+#include <iterator>
+
+// check for spaceship support
+#if defined __has_include
+#    if __has_include(<version>)
+#        include <version>
+#    endif
+#    if __has_include(<compare>)
+#        include <compare>
+#        if defined(__cpp_lib_three_way_comparison) && defined(__cpp_impl_three_way_comparison)
+#            define USESPACESHIP
+#        endif
+#    endif
+#endif
+
+#if __has_cpp_attribute(nodiscard)
+#    define NODISCARD [[nodiscard]]
+#else
+#    define NODISCARD
+#endif
+
+namespace vikunja
+{
+    namespace mem
+    {
+        namespace iterator
+        {
+            /**
+             * @brief A constant iterator, returning a value given initially at any index. As such it has no bounds,
+             * other than the bounds of the index type used.
+             * @tparam DataType The type of the data
+             * @tparam IdxType The type of the index
+             */
+            template<typename DataType, typename IdxType = int64_t>
+            class ConstantIterator
+            {
+            public:
+                // Need all 5 of these types for iterator_traits
+                using difference_type = IdxType;
+                using value_type = DataType;
+                using pointer = DataType*;
+                using reference = DataType&;
+                using iterator_category = std::random_access_iterator_tag;
+
+                /**
+                 * @brief Constructor for the ConstantIterator
+                 * @param value The value to initialize the iterator with
+                 * @param idx The index for the iterator, default 0
+                 */
+                ConstantIterator(const DataType& value, const IdxType& idx = static_cast<IdxType>(0))
+                    : v(value)
+                    , index(idx)
+                {
+                }
+
+                /**
+                 * @brief Dereference operator to receive the stored value
+                 */
+                NODISCARD ALPAKA_FN_INLINE const DataType& operator*() const
+                {
+                    return v;
+                }
+
+                /**
+                 * @brief Index operator to get stored value at some given offset from this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE const DataType& operator[](int) const
+                {
+                    return v;
+                }
+
+#pragma region arithmeticoperators
+                /**
+                 * @brief Postfix increment operator
+                 * @note Use prefix increment operator instead if possible to avoid copies
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator++()
```

Comment:
```suggestion
                ALPAKA_FN_INLINE ConstantIterator operator++()
```
Incrementing an iterator without using the old value is a common use in my opinion. No need for a warning here.

Reply:
Oh very true, I just mindlessly put Nodiscard everywhere...